src/clients/Deluge.ts

```diff
@@ -0,0 +1,225 @@
+import { Searchee } from "../searchee.js";
+import { TorrentClient } from "./TorrentClient.js";
+
+export default class Deluge implements TorrentClient {
+	private msgId = 0;
+	private loggedIn = false;
+	private delugeCookie = "";
+	private delugeWebUrl: string;
+	private delugeLabel: string;
+	private delugeWebPassword: string;
+	private enLabeled: boolean;
+
+	constructor() {
+		this.delugeWebPassword = getRuntimeConfig().delugeWebPassword;
+		this.delugeWebUrl = getRuntimeConfig().delugeWebUrl;
+		this.delugeLabel = getRuntimeConfig().dataCategory;
+		this.enLabeled = false;
+	}
+
+	/**
+	 * validates the login and host for deluge webui
+	 */
+	async validateConfig(): Promise<void> {
+		const result = await this.authenticate();
+		if (!result) {
+			logger.verbose({
+				label: Label.DELUGE,
+				message: "failed to authenticate/reach the deluge daemon.",
+			});
+			throw new CrossSeedError(
+				`Could not authenticate/reach Deluge: ${this.delugeWebUrl}`
+			);
+		}
+	}
+
+	/**
+	 * connects and authenticates to the webui
+	 */
+	private async authenticate(): Promise<boolean> {
+		if (!this.loggedIn) {
+			let response = await this.call({
+				params: [this.delugeWebPassword],
+				method: "auth.login",
+			});
+			if (
+				response &&
+				response.headers &&
+				response["data"]["result"] &&
+				response.headers["set-cookie"]
+			) {
+				this.delugeCookie =
+					response.headers["set-cookie"][0].split(";")[0];
+				this.loggedIn = true;
+			}
+		} else {
+			let response = await this.call({
+				params: [this.delugeCookie],
+				method: "auth.check_session",
+			});
+			this.loggedIn = !!response;
+		}
+		return this.loggedIn;
+	}
+
+	/**
+	 * authenticates and sends JSON-RPC calls to deluge
+	 */
+	private async sendCall(method: string, params: any[]) {
+		let response;
+		if (!this.loggedIn) {
+			await this.authenticate();
+			response = await this.call({
+				params,
+				method,
+			});
+		} else
+			response = await this.call({
+				params: params,
+				method,
+			});
+
+		return response && response.data && response.data.error === null
+			? response.data
+			: response.data.error;
+	}
+
+	private async call(body: any): Promise<AxiosResponse | null> {
+		return new Promise((resolve) => {
+			body.id = ++this.msgId;
+			if (this.msgId > 1024) this.msgId = 0;
+			axios
+				.post(this.delugeWebUrl, body, {
+					headers: {
+						"Content-Type": "application/json",
+						Cookie: this.delugeCookie,
+					},
+				})
+				.then((data) => {
+					resolve(data);
+				})
+				.catch(() => resolve(null));
+		});
+	}
+
+	/**
+	 * lists plugins and adds/sets labels
+	 * returns true if successful.
+	 */
+	private async labelEnabled() {
+		const enabledLabels = await this.sendCall(
+			"core.get_enabled_plugins",
+			[]
+		);
+		if (enabledLabels["result"].includes("Label")) {
+			return true;
+		}
+		return false;
+	}
+	private async setLabel(torrenthash: string) {
+		if (this.enLabeled) {
```

Comment:
I don't see anything that will set this to true

Reply:
Another oversight...will fix this :P